onnxruntime/core/providers/dml/dml_provider_factory.cc

```diff
@@ -141,13 +142,19 @@ std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_DML(in
 
 }  // namespace onnxruntime
 
+// These exports should be deprecated, and the exports from the OrtDmlApi should be used
+// to provide reliably versioned provider API exports.
+// [[deprecated]]
```

Comment:
> [[deprecated]] [](http://example.com/codeflow?start=2&length=15)

Since ORT on C++17, can we try using that somehow?

Reply:
Putting either a C++ attribute or //comment in the header would be more visible to people than the cpp, I expect, as they're more likely to look at the header prototype than the implementation cpp (e.g. when you press F12 in Visual Studio for "goto definition", it takes you to the header first).

Reply:
Moved into header per @fdwr. I think we should eventually deprecate it. I dont know if we are ready to do that with this change yet though.

Reply:
Sounds fine to me. version.next++.